source/agora/test/EnrollmentManager.d

```diff
@@ -221,3 +221,188 @@ unittest
     // Check if a new pre-image has been revealed from the restarted node
     assert(preimage_2.isValid(org_preimage, validator_cycle));
 }
+
+/// Situation: One misbehaving node sends an Enrollment for an already-enrolled
+///            validator.
+/// Expectation: The nomination is rejected.
+unittest
+{
+    import agora.common.Config;
+    import agora.common.Task;
+    import agora.common.crypto.Key;
+    import agora.consensus.protocol.Nominator;
+    import agora.consensus.data.ConsensusData;
+    import agora.node.Ledger;
+    import agora.node.Validator;
+    import agora.network.NetworkManager;
+
+    import geod24.Registry;
+
+    static class BadNominator : Nominator
+    {
+        private Ledger ledger;
+        private bool is_nominating;
+
+        /// Ctor
+        public this (NetworkManager network, KeyPair key_pair,
+            Ledger ledger, TaskManager taskman)
+        {
+            this.ledger = ledger;
+            super(network, key_pair, ledger, taskman);
+        }
+
+        ///
+        public override bool isNominating () @safe @nogc nothrow
+        {
+            return this.is_nominating;
+        }
+
+        ///
+        public override void tryNominate () @safe
+        {
+            // Checks done in `Nominator.tryNominate`
```

Comment:
I would change the comment to say: "// Most of the code, save for the line commented as such, are taken from `Validator.tryNominate`"

Or something similar